ipapython/dnsutil.py

```diff
@@ -374,10 +380,7 @@ def check_zone_overlap(zone, raise_on_error=True):
                          zone, e)
             ns = []
 
-        msg = u"DNS zone {0} already exists in DNS".format(zone)
-        if ns:
-            msg += u" and is handled by server(s): {0}".format(', '.join(ns))
-        raise ValueError(msg)
+        raise DNSZoneAlreadyExists(zone.to_text(), ns)
```

Comment:
This does not result in a proper error message. You must pass the arguments as keyword arguments:

```
>>> import dns.exception
>>> class DNSZoneAlreadyExists(dns.exception.DNSException):
...     supp_kwargs = {'zone', 'ns'}
...     fmt = (u"DNS zone {zone} already exists in DNS "
...            "and is handled by server(s): {ns}")
... 
>>> DNSZoneAlreadyExists("zone", "ns")
DNSZoneAlreadyExists('zone', 'ns')
>>> str(DNSZoneAlreadyExists("zone", "ns"))
"('zone', 'ns')"
>>> DNSZoneAlreadyExists("zone", "ns").msg is None
True
```